include/mata/nfa.hh

```diff
@@ -196,22 +221,39 @@ struct Post : private Util::OrdVector<Move> {
 
     virtual ~Post() = default;
 
-    const_iterator find(const Move& m) const override { return Util::OrdVector<Move>::find(m);}
-    iterator find(const Move& m) override { return Util::OrdVector<Move>::find(m);}
+    //Trying to make Past movable, oes it make sense?
+    Post(Post&& rhs) = default;
+    Post(const Post& rhs) = default;
+    Post & operator=(Post&& rhs) = default;
+    Post & operator=(const Post& rhs) = default;
+
+    inline const_iterator find(const Move& m) const override { return Util::OrdVector<Move>::find(m);}
+    inline iterator find(const Move& m) override { return Util::OrdVector<Move>::find(m);}
+
+    void inline insert(const Move& m) override { Util::OrdVector<Move>::insert(m); }
+
+    // dangerous, breaks the sortedness invariant
+    void inline push_back(const Move& m) override { Util::OrdVector<Move>::push_back(m); }
+
+    inline const Move& back() const override { return Util::OrdVector<Move>::back(); }
 
-    void insert(const Move& m) override { Util::OrdVector<Move>::insert(m); }
+    // is adding this non-const version ok?
+    inline Move& back() { return Util::OrdVector<Move>::back(); }
 
-    const Move& back() const override { return Util::OrdVector<Move>::back(); }
+    inline void resize(size_t size) override { Util::OrdVector<Move>::resize(size); }
 
     void remove(const Move& m)  { Util::OrdVector<Move>::remove(m); }
 
     bool empty() const override{ return Util::OrdVector<Move>::empty(); }
     size_t size() const override { return Util::OrdVector<Move>::size(); }
 
-	const std::vector<Move>& ToVector() const
-	{
-		return Util::OrdVector<Move>::ToVector();
-	}
+    const std::vector<Move>& ToVector() const { return Util::OrdVector<Move>::ToVector(); }
+
+    //Could we somehow use the && thingy here? It is supposed to be faster?
```

Comment:
Already there, remove the comment after verifying that the function does what it should and that direct forwarding is indeed faster. Is it not enough to pass a const reference?

Reply:
I don't know, my impression was that && is not slower, so I would just use as the befault.